src/notebooklm/_artifacts.py

```
@@ -2472,3 +2473,198 @@ def _is_media_ready(self, art: builtins.list[Any], artifact_type: int) -> bool:
+        elif artifact_type == ArtifactType.QUIZ:
+            fmt = output_format or "json"
+            return await self.download_quiz(notebook_id, output_path, artifact_id, fmt)
+        elif artifact_type == ArtifactType.FLASHCARDS:
+            fmt = output_format or "json"
+            return await self.download_flashcards(notebook_id, output_path, artifact_id, fmt)
+        elif artifact_type == ArtifactType.DATA_TABLE:
+            return await self.download_data_table(notebook_id, output_path, artifact_id)
+        elif artifact_type == ArtifactType.MIND_MAP:
+            return await self.download_mind_map(notebook_id, output_path, artifact_id)
+        else:
+            raise ArtifactError(f"No download method for artifact type: {artifact_type}")
```

Ruff (0.15.9)

[error] Undefined name `ArtifactError`

(F821)